sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/StringExpressionsSuite.scala

```diff
@@ -1124,8 +1124,9 @@ class StringExpressionsSuite extends SparkFunSuite with ExpressionEvalHelper {
     ).foreach { case (str: String, format: String) =>
       val toNumberExpr = ToNumber(Literal(str), Literal(format))
       assert(toNumberExpr.checkInputDataTypes() == TypeCheckResult.TypeCheckSuccess)
-      checkExceptionInExpression[IllegalArgumentException](
-        toNumberExpr, "does not match the given number format")
+
+      checkExceptionInExpression[SparkException](
+        toNumberExpr, "[INTERNAL_ERROR] Undefined error message parameter")
```

Comment:
Sorry for late responding, @MaxGekk .

I've been thinking about if there is better way to test expression with error class here.

Please let me know if there is better way to test this ? 🙏

Reply:
See https://github.com/apache/spark/pull/38177#pullrequestreview-1153416983

Reply:
Seems like I should also fix the test & generated golden files.

Let me fix soon!